internal/pkg/wirecli/wirek8s/wirek8s.go

```diff
@@ -73,36 +90,62 @@ func NewRestConfig(master clik8s.MasterURL, path clik8s.KubeConfigPath) (*rest.C
 	return clientcmd.BuildConfigFromFlags(string(master), string(path))
 }
 
+// NewPluginConfig returns a new PluginConfig
+func NewPluginConfig() *types.PluginConfig {
+	return plugin.DefaultPluginConfig()
+}
+
+// NewResMapFactory returns a rew ResMap Factory
+func NewResMapFactory(pc *types.PluginConfig) *resmap.Factory {
+	uf := kunstruct.NewKunstructuredFactoryWithGeneratorArgs(
+		&types.GeneratorMetaArgs{
+			PluginConfig: pc,
+		})
+	return resmap.NewFactory(resource.NewFactory(uf))
+}
+
+// NewTransformerFactory returns a new transformer factory
+func NewTransformerFactory() transformer.Factory {
+	return ktransformer.NewFactoryImpl()
+}
+
+// NewFileSystem returns a new filesystem
+func NewFileSystem() fs.FileSystem {
+	return fs.MakeRealFS()
+}
+
+// NewConfigProvider returns a new ConfigProvider
+func NewConfigProvider(rf *resmap.Factory, fSys fs.FileSystem, tf transformer.Factory, pc *types.PluginConfig) resourceconfig.ConfigProvider {
```

Comment:
Return KustomizeProvider, and bind ConfigProvider to KustomizeProvider

Reply:
Would a struct provider be more appropriate here ?
https://github.com/google/wire/blob/master/docs/guide.md#struct-providers